06_puppy/hsy3418/mapStore.go

```diff
@@ -0,0 +1,49 @@
+package main
+
+import "fmt"
+
+// MapStore is a implementation of storer for the storage of puppies
+type MapStore struct {
+	puppies map[int32]Puppy
+}
+
+func NewMapStore() *MapStore {
+	return &MapStore{
+		puppies: map[int32]Puppy{},
+	}
+}
+
+// CreatePuppy adds a nuw puppy to the puppies store
+func (m *MapStore) CreatePuppy(puppy Puppy) error {
+	if _, ok := m.puppies[puppy.ID]; !ok {
```

Comment:
`exists` would be better instead of `ok`

Reply:
@kasunfdo this a golang this: [comma ok idiom](https://golang.org/doc/effective_go.html#maps)
however, in my opinion you typically keep the error path inside the if so use `if  _, ok := m.puppies[puppy.ID]; ok { return err }`